src/Behat/Config/Formatter/Formatter.php

```diff
@@ -55,8 +78,63 @@ public function toArray(): array
         return $this->settings;
     }
 
+    public function toPhpExpr(): Expr
+    {
+        $formatterObject =  $this->builderFactory->new(new FullyQualified(self::class));
+
+        $expr = $this->applyBaseSettings($formatterObject);
+
+        if ($this->settings === []) {
+            $args = $this->builderFactory->args([$this->name]);
+        } else {
+            $args = $this->builderFactory->args([$this->name, $this->settings]);
+        }
+        $formatterObject->args = $args;
+
+        return $expr;
+    }
+
     public static function defaults(): array
     {
         return [];
     }
+
+    public function applyBaseSettings(Expr $expr): Expr
+    {
+        foreach ($this->settings as $settingName => $setting) {
```

Comment:
This function is used to apply the base options like `output_path` by calling the corresponding function (`withOutputPath()` for example)